src/main/javascript/crypto/src/libs/IdentifierAttestation.ts

```diff
@@ -0,0 +1,36 @@
+import {KeyPair} from "./KeyPair";
+import {AttestationCrypto} from "./AttestationCrypto";
+import {Asn1Der} from "./DerUtility";
+
+export class IdentifierAttestation {
+    private crypto: AttestationCrypto;
+    private version: string;
+    static OID_OCTETSTRING = "1.3.6.1.4.1.1466.115.121.1.40";
+    constructor(private riddle: Uint8Array, private keys: KeyPair) {
+        this.crypto = new AttestationCrypto();
+        this.setVersion(18); // Our initial version
+        // this.setSubject("CN=" + this.crypto.addressFromKey(this.keys));
+        // SEQUENCE (1 elem)
+        // OBJECT IDENTIFIER
+        // this.setSignature(AttestationCrypto.OID_SIGNATURE_ALG);
+        // try {
+        //     SubjectPublicKeyInfo spki = SubjectPublicKeyInfoFactory.createSubjectPublicKeyInfo(key);
+        //     super.setSubjectPublicKeyInfo(spki);
+        // } catch (IOException e) {
+        //     throw new RuntimeException(e);
+        // }
+        this.setRiddle(riddle);
+    }
+    setVersion(version: number) {
+        this.version = Asn1Der.encodeAsInteger( BigInt(version) );
+    }
+    // setSubject(subject: string) {
+    //     this.version = Asn1Der.encode('INTEGER', subject);
+    // }
+    // setSignature(signature: string) {
+    //     this.version = Asn1Der.encode('INTEGER', signature);
+    // }
+    setRiddle(riddle: Uint8Array) {
```

Comment:
It seems that the validity checks are not done in this class.

Reply:
I removed this file. we dont need it in JS at the moment. Tell me If you need me to implement this class too.

Reply:
The checks being done in IdentifierAttestation.checkValidity are needed when we want to use Identity attestations (which are currently the only kind we are using). For this reason they could simply be done in the Attestation object, but to keep things more dynamic I would prefer if it was done in this class, if it is not too much trouble.

Reply:
IdentifierAttestation.checkValidity added